download_from_csv.py

```diff
@@ -1,41 +1,163 @@
-'''Script to download your spotify playlists
+'''
+Script to download your spotify playlists:
 
-Convert your spotify playlists to csv from here: http://joellehman.com/playlist/
+Convert your spotify playlist to csv from here: http://joellehman.com/playlist
+or provide playlist URI, your username and authorization token here
 and then download all the songs through this script
 Depends upon youtube_dl, eyed3 and unidecode pip packages
 '''
 
+from __future__ import unicode_literals
 import os
 import csv
-import eyed3
 import argparse
+import json
+import UserString
+
+import eyed3
 import youtube_dl
 from unidecode import unidecode
-from __future__ import unicode_literals
+import requests
+import wget
+from bs4 import BeautifulSoup
+
+
+# Youtube-dl options
+opts = {
+    'format': 'bestaudio/best',
+    'forcejson': True,
+    'postprocessors': [{
+        'key': 'FFmpegExtractAudio',
+        'preferredcodec': 'mp3',
+        'preferredquality': '256',
+    }],
+    'progress_hooks': '',
+    'logger': '',
+    'outtmpl': ''
+}
+# user-agents
+headers = {
+    'User-Agent': 'Mozilla/5.0'
+}
+
+
+class MyLogger(object):
+
+    def debug(self, msg):
+        pass
+
+    def warning(self, msg):
+        pass
 
+    def error(self, msg):
+        print(msg)
+
+
+def download_finish(d):
+    if d['status'] == 'finished':
+        print '\x1b[1A\x1b[2K'
+        print "\x1b[1A[\033[93mConverting\033[00m] %s" % d['filename']
+
+
+def download_image(keywords):
```

Comment:
1. Instead of taking the `keywords`, take the song dictionary as argument
2. `<album> <artist>` query gives better results than your current keywords
3. Move this function to separate module called `album_art` and change its name to just `download` (More readable and modular)
4. Instead of saving the image to disk (using wget), use requests to fetch the image (it will also remove dependency on wget) and return the stream, which can then be directly used in `afile.tag.images.set()`